warehouse/query-core/src/main/java/datawave/query/jexl/visitors/EventDataQueryExpressionVisitor.java

```diff
@@ -151,17 +153,77 @@ public boolean apply(Key key) {
+                                normalizedPatternMatchers.add(fieldPatterns.get(fieldPattern));
+                            }
+                        } catch (Exception e) {
+                            // can't normalize this pattern, add the original matcher
+                            normalizedPatternMatchers.add(fieldPatterns.get(fieldPattern));
+                        }
+                    }
+                    
+                    // normalize all values
+                    for (String fieldValue : fieldValues) {
+                        try {
+                            String normalizedValue = type.normalize(fieldValue);
+                            if (normalizedValue != null) {
+                                normalizedFieldValues.add(normalizedValue);
+                            } else {
+                                // can't normalize this value, add the original
+                                normalizedFieldValues.add(fieldValue);
+                            }
+                        } catch (Exception e) {
+                            // can't normalize this value, add the original
+                            normalizedFieldValues.add(fieldValue);
+                        }
+                    }
+                    
+                    // normalize all ranges
+                    for (LiteralRange range : fieldRanges) {
+                        try {
+                            LiteralRange normalizedRange = new LiteralRange(range.getFieldName(), range.getNodeOperand());
+                            
+                            String normalizedLower = type.normalize(range.getLower().toString());
+                            String normalizedUpper = type.normalize(range.getUpper().toString());
+                            
+                            if (normalizedLower != null && normalizedUpper != null) {
+                                normalizedRange.updateLower(normalizedLower, range.isLowerInclusive());
+                                normalizedRange.updateUpper(normalizedUpper, range.isUpperInclusive());
+                                
+                                normalizedRanges.add(normalizedRange);
+                            } else {
+                                // can't normalize the range values, add the original
+                                normalizedRanges.add(range);
+                            }
+                        } catch (Exception e) {
+                            // can't normalize the range values, add the original
+                            normalizedRanges.add(range);
+                        }
+                    }
+                }
                 
-                for (String normalizedFieldValue : normalizedFieldValues) {
-                    if (fieldValues.contains(normalizedFieldValue)) {
+                Set<String> fieldValuesToEvaluate = EventDataQueryExpressionVisitor.extractNormalizedValues(types);
```

Comment:
We likely need to add the non-normalized value to this set as well, see the comment in #246